fidesctl/src/fidesctl/core/generate_dataset.py

```diff
@@ -116,35 +134,143 @@ def create_dataset_collections(
+                )
+    return uncategorized_fields, total_field_count
+
+
+def find_all_uncategorized_dataset_fields(
+    manifest_taxonomy: Optional[Taxonomy],
+    db_collections: Dict[str, Dict[str, List[str]]],
+    url: AnyHttpUrl,
+    headers: Dict[str, str],
+) -> Tuple[List[str], int]:
     """
-    Generate a partial dataset manifest, sans tables/fields,
-    given a database engine.
+    Finds all uncategorized fields given an database modeled object. Datasets
+    are pulled from the server unless a manifest taxonomy is supplied.
     """
-    url = engine.url
-    name = url.database
-    dataset = Dataset(
-        fides_key=name,
-        name=name,
-        description=f"Fides Generated Description for Dataset: {name}",
-        collections=collections,
+    uncategorized_fields = []
+    total_field_count = 0
+    for db_dataset_key in db_collections.keys():
+        dataset = (
+            next(
+                (
+                    dataset
+                    for dataset in manifest_taxonomy.dataset
+                    if dataset.fides_key == db_dataset_key
+                ),
+                None,
+            )
+            if manifest_taxonomy
+            else get_server_resource(
+                url=url,
+                resource_type="dataset",
+                resource_key=db_dataset_key,
+                headers=headers,
+            )
+        )
+        db_dataset = db_collections.get(db_dataset_key, {})
+        (
+            current_uncategorized_keys,
+            current_field_count,
+        ) = find_uncategorized_dataset_fields(
+            dataset_key=db_dataset_key, dataset=dataset, db_dataset=db_dataset
+        )
+        total_field_count += current_field_count
+        uncategorized_fields += current_uncategorized_keys
+
+    return uncategorized_fields, total_field_count
+
+
+def database_coverage(
+    connection_string: str,
+    manifest_dir: Optional[str],
+    coverage_threshold: int,
+    url: AnyHttpUrl,
+    headers: Dict[str, str],
+) -> None:
+    """
+    Given a database connection string, fetches collections
+    and fields and compares them to existing datasets or datasets in a
+    local manifest (if one is provided).
+
+    Prints uncategorized fields and raises an exception if coverage
+    is lower than provided threshold.
+    """
+    manifest_taxonomy = parse(manifest_dir) if manifest_dir else None
+
+    # Generate the collections and fields for the target database
+    db_engine = get_db_engine(connection_string)
+    db_collections = get_db_collections_and_fields(db_engine)
+
+    uncategorized_fields, db_field_count = find_all_uncategorized_dataset_fields(
+        manifest_taxonomy=manifest_taxonomy,
+        db_collections=db_collections,
+        url=url,
+        headers=headers,
     )
-    return dataset
+    coverage_percent = (
+        int(((db_field_count - len(uncategorized_fields)) / db_field_count) * 100)
+        if db_field_count > 0
+        else 100
```

Comment:
yeah if someone is explicitly checking a database and it comes back empty, that seems like a problem and we'd want to surface that

Reply:
I added this within the same method for now. Adding it to a method like `find_all_uncategorized_dataset_fields` feels bad since these methods could be used for other purposes.